Applications/Spire/Source/Ui/RowViewListModel.cpp

```diff
@@ -0,0 +1,84 @@
+#include "Spire/Ui/RowViewListModel.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+
+RowViewListModel::RowViewListModel(std::shared_ptr<TableModel> source, int row)
+    : m_source(std::move(source)),
+      m_row(row) {
+  if(row < 0 || row >= m_source->get_row_size()) {
+    m_row = none;
```

Comment:
When ```m_row == -1```, do the following:

    m_source_connection.disconnect();
    m_source = nullptr;